# **Estudo Individual: Atividade para casa**
**Objetivo:** *Implementar o primeiro modelo de ML e entender o fluxo básico.*

###**Instruções:**

**1.** Baixar o notebook exemplo

**2.** Executar o código de regressão linear (dataset de preço de casas)

**3.** Testar mudanças:
- Alterar a proporção de treino/teste (ex: 70/30, 90/10)
- Tentar um dataset diferente (se quiser)

**4.** Responder no notebook:
- Qual foi o valor do R2 obtido?
- O que esse valor significa?
- O que aconteceu quando você mudou a proporção de
treino/teste?

**5.** Subir o notebook respondido na pasta da semana 8 do repositório



---



# **Regressão Linear — Preço de Casas**

## **Importação das bibliotecas**

In [13]:
import pandas as pd

from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score

##**Carregar dataset**

In [14]:
df = pd.read_csv("kc_house_data.csv")

print(df.head())

           id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     lo

##**Explorar os dados**

In [15]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

In [16]:
print(df.describe())

                 id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517    

##**Escolher variáveis**
Previsão do preço (price) usando:
- quartos
- banheiros
- área da casa

In [17]:
X = df[["bedrooms", "bathrooms", "sqft_living"]]

y = df["price"]

##**Separar treino e teste**
Primeiro teste: 80/20

In [18]:
X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42
)

##**Criar modelo**

In [19]:
modelo = LinearRegression()

##**Treinar modelo**

In [20]:
modelo.fit(X_train, y_train)

LinearRegression()

##**Fazer previsões**

In [21]:
previsoes = modelo.predict(X_test)

##**Avaliar modelo com R²**

In [22]:
r2 = r2_score(y_test, previsoes)

print("R2:", r2)

R2: 0.50893471722649


##**Testar outras proporções**
**Teste 70/30**

In [23]:
X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.3,
    random_state=42
)

modelo = LinearRegression()

modelo.fit(X_train, y_train)

previsoes = modelo.predict(X_test)

r2 = r2_score(y_test, previsoes)
print("R2 com 70/30:", r2)

R2 com 70/30: 0.49782014676842645


**Teste 90/10**

In [24]:
X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.1,
    random_state=42
)

modelo = LinearRegression()

modelo.fit(X_train, y_train)

previsoes = modelo.predict(X_test)

r2 = r2_score(y_test, previsoes)

print("R2 com 90/10:", r2)

R2 com 90/10: 0.47860082707886675




---



##**Pergunta 1:** *Qual foi o valor do R² obtido?*

O modelo apresentou os seguintes valores de R²:

- **80/20** → 0.5089
- **70/30** → 0.4978
- **90/10** → 0.4786

##**Pergunta 2:** *O que esse valor significa?*

- O R² indica o quanto o modelo consegue explicar a variação do preço das casas utilizando as variáveis escolhidas.

- Os valores ficaram próximos de 0.5, o que significa que o modelo consegue explicar aproximadamente 50% da variação dos preços.

- Isso mostra que o modelo possui desempenho moderado, mas ainda existem muitos fatores que influenciam o preço das casas e não foram considerados.

##**Pergunta 3:** *O que aconteceu quando mudou treino/teste?*

- Ao alterar a proporção entre treino e teste, os valores de R² sofreram pequenas mudanças.

- O modelo apresentou melhor desempenho com a divisão 80/20. Já na divisão 90/10, o R² diminuiu um pouco, possivelmente porque o conjunto de teste ficou muito pequeno, tornando a avaliação menos estável.

- Essas diferenças mostram que a proporção treino/teste pode influenciar na capacidade de generalização do modelo.